test/integration/mxGraph.model.style.api.test.ts

```diff
@@ -545,6 +545,81 @@ describe('mxGraph model - update style', () => {
     });
   });
 
+  describe('Message flow icons', () => {
+    beforeEach(() => {
+      bpmnVisualization.load(readFileSync('../fixtures/bpmn/registry/message-flows-with-and-without-icon.bpmn'));
+    });
+
+    const stroke: Stroke = { color: 'pink', opacity: 72, width: 7 };
+    const opacity = 84;
+
+    test('Update the style of a message flow without icon', () => {
+      bpmnVisualization.bpmnElementsRegistry.updateStyle('MessageFlow_1', { opacity, stroke });
+      expect('MessageFlow_1').toBeMessageFlow({
+        messageVisibleKind: MessageVisibleKind.NONE,
+        opacity,
+        stroke,
+        // not under test
+        verticalAlign: 'bottom',
+      });
+    });
+
+    test('Update the style of a message flow with icon (initiating)', () => {
+      bpmnVisualization.bpmnElementsRegistry.updateStyle('MessageFlow_2_msgVisibilityKind_initiating', { opacity, stroke });
+      expect('MessageFlow_2_msgVisibilityKind_initiating').toBeMessageFlow({
+        messageVisibleKind: MessageVisibleKind.INITIATING,
+        stroke,
+        opacity,
+        // not under test
+        verticalAlign: 'bottom',
+      });
+    });
+
+    test('Update the properties of a message flow with icon (non initiating) and restore its default values', () => {
+      // Check that the element uses default values
+      expect('MessageFlow_3_msgVisibilityKind_non_initiating').toBeMessageFlow({
+        messageVisibleKind: MessageVisibleKind.NON_INITIATING,
+        // not under test
+        verticalAlign: 'bottom',
+      });
+
+      bpmnVisualization.bpmnElementsRegistry.updateStyle('MessageFlow_3_msgVisibilityKind_non_initiating', {
+        opacity,
+        stroke,
+      });
+
+      // Check that the style has been updated
+      expect('MessageFlow_3_msgVisibilityKind_non_initiating').toBeMessageFlow({
+        messageVisibleKind: MessageVisibleKind.NON_INITIATING,
+        opacity,
+        stroke,
+        // not under test
+        verticalAlign: 'bottom',
+      });
+
+      // Reset the style by passing special values
+      bpmnVisualization.bpmnElementsRegistry.updateStyle('MessageFlow_3_msgVisibilityKind_non_initiating', <EdgeStyleUpdate>{
+        font: {
+          color: 'default',
+          opacity: 'default',
+        },
```

Comment:
**thought:** we haven't set the font style before. We should probably do it to fully verify the style update and reset

Reply:
done in 61ca7935